### PR TITLE
Removing runit

### PR DESCRIPTION
**Pull request by hunter**
It seems we've had a few issues with runit recently and thought I'd start a discussion (with code) for switching it out for something else.

We've used forego (a go based foreman replacement) in other projects so decided to test it out an alternative for launching multiple processes in ceph-docker. It's currently untested but thought I'd put it out there first before spending more time getting it ready.

The main change is that forego will run the processes specified in a 'Procfile' which is generated slightly differently to the runit. Overall there are slightly fewer dependencies with a single binary that will work across all base OSes. Logs are still sent to stdout/err with each line prefixed with the process name (cluster name and osd ID).

Thoughts? Not a visible change but may simplify things slightly

### CODE REVIEW

File: ceph-releases/jewel/ubuntu/14.04/daemon/entrypoint.sh [hunk 1]
```diff
@@ -617,9 +612,12 @@ function osd_disks {
         echo "Device ${OSD_DEV} is corrupt for /var/lib/ceph/osd/${CLUSTER}-${OSD_ID}"
         exit 1
       fi
-      set_osd_run ${CLUSTER} ${OSD_ID}
+
+			echo "${CLUSTER}-${OSD_ID}: /usr/bin/ceph-osd ${CEPH_OPTS} -f -d -i ${OSD_ID} --setuser ceph --setgroup disk" | tee -a /etc/forego/${CLUSTER}/Procfile
```
Comment (leseb) [inline ceph-releases/jewel/ubuntu/14.04/daemon/entrypoint.sh]: what's that indent?

File: ceph-releases/jewel/ubuntu/14.04/daemon/entrypoint.sh [hunk 2]
```diff
@@ -617,9 +612,12 @@ function osd_disks {
...
+
     done
-    exec /sbin/my_init
+
+		exec /usr/local/bin/forego start -f /etc/forego/${CLUSTER}/Procfile
```
Comment (leseb) [inline ceph-releases/jewel/ubuntu/14.04/daemon/entrypoint.sh]: ditto

File: ceph-releases/jewel/ubuntu/14.04/daemon/entrypoint.sh [hunk 3]
```diff
@@ -652,9 +650,14 @@ function osd_disks {
           echo "OSD (PID ${OSD_PID}) is running, waiting till it exits"
           while [ -e /proc/${OSD_PID} ]; do sleep 1;done
       fi
-      set_osd_run ${CLUSTER} ${OSD_ID}
+
+			echo "${CLUSTER}-${OSD_ID}: /usr/bin/ceph-osd ${CEPH_OPTS} -f -d -i ${OSD_ID} --setuser ceph --setgroup disk" | tee -a /etc/forego/${CLUSTER}/Procfile
```
Comment (leseb) [inline ceph-releases/jewel/ubuntu/14.04/daemon/entrypoint.sh]: ditto

File: ceph-releases/jewel/ubuntu/14.04/daemon/entrypoint.sh [hunk 4]
```diff
@@ -652,9 +650,14 @@ function osd_disks {
...
+
+
     done
-    exec /sbin/my_init
+
+
+		exec /usr/local/bin/forego start -f /etc/forego/${CLUSTER}/Procfile
```
Comment (leseb) [inline ceph-releases/jewel/ubuntu/14.04/daemon/entrypoint.sh]: ditto

File: ceph-releases/jewel/ubuntu/14.04/daemon/entrypoint.sh (hunk 1 above)
Comment (hunter) [inline ceph-releases/jewel/ubuntu/14.04/daemon/entrypoint.sh]: Not quite sure actually. Something got messed up along the way

**Comment by leseb** [comment]
This looks like an interesting change, I'm not strongly against it, it's true that the binary makes things much easier so we don't have to find dependancies for all distros. The CI seems to be failing building the image though, see https://travis-ci.org/ceph/ceph-docker/builds/165453217#L1322

File: ceph-releases/jewel/ubuntu/14.04/daemon/entrypoint.sh (hunk 1 above)
Comment (leseb) [inline ceph-releases/jewel/ubuntu/14.04/daemon/entrypoint.sh]: Thanks!

**Comment by hunter** [comment]
Made a few changes to fix the build errors and rogue tabs. If this is something we want to use then I can update any missing entrypoints (can't remember if I got everything)

**Comment by leseb** [comment]
@hunter Looks good to me, sticking with Jewel is fine for most of the new changes.

**Pull request merged**